### PR TITLE
#149 Use password hash for login

### PR DESCRIPTION
**Pull request by CarlosNZ**
Implements #149. As discussed in that issue, I've left the "password plugin" to a [separate issue](#253) as there are additional complexities involved with that.

Requires [back-end PR #173](https://github.com/openmsupply/application-manager-server/pull/173) -- most of the logic changes are in there. This front-end is basically just naming/tidying of existing code.

Logging in with one of the existing users (password: "123456") should work with hash comparison on back-end.

### CODE REVIEW

File: src/containers/User/Login.tsx [hunk 1]
```diff
@@ -17,8 +17,8 @@ const Login: React.FC = () => {
 
   const handleSubmit = async (event: any) => {
     event.preventDefault()
-    const passwordHash = hashPassword(password)
-    const loginResult = await attemptLogin(username, passwordHash)
+    const loginResult = await attemptLogin(username, password)
+    console.log('Login Result', loginResult)
```
Comment (nmadruga) [inline src/containers/User/Login.tsx:21]: I think this can be removed now

**Review by nmadruga — CHANGES_REQUESTED** [review]
Nice and tidy, I'm having a look on the back-end now.

For now can I just ask something? Can you please also add the encrypted password replace `const hardcodedPassword = '123456'` on the `UserSelection` container so we can continue to login with different test users.

**Comment by CarlosNZ** [comment]
> For now can I just ask something? Can you please also add the encrypted password replace `const hardcodedPassword = '123456'` on the `UserSelection` container so we can continue to login with different test users.

This shouldn't need changing -- the password is sent to the server, which compares its hash against the stored one. I think you'll find it's still working okay, no?

Or did you mean something else?

**Comment by nmadruga** [comment]
> This shouldn't need changing -- the password is sent to the server, which compares its hash against the stored one. I think you'll find it's still working okay, no?

Yes! Sorry, what a confusion/ That's all good! I'll test again and see if this can be already approved

**Review by nmadruga — APPROVED** [review]

**Pull request merged**